### ISSUE TITLE
How do I open Image Dialog programmatically?

### ISSUE THREAD

**Issue by kanertuncel** [GetmeUK/ContentTools#334]
Hi @anthonyjb , 
It works like a charm, thank you for this greate tool. I searched but I couldn't find out a solution. How do I  open Image Dialog programmatically? I need to open it when user double clicks on existing image. Thanks in advance!

**Comment by anthonyjb** [comment]
Hi @kanertuncel,

So there's two parts to this I guess, firstly capturing the double click event against an image element and secondly triggering the image tool:

```JavaScript
ContentEdit.Root.get().bind('mount', function (element) {

    // We're only interested in images
    if (element.type() !== 'Image') {
        return;
    }

    // Capture a double-click event against the image element
    element.domElement().addEventListener('dblclick', function (ev) {
        ContentTools.Tools.Image.apply(element);
    });
});
```

> I haven't had chance to test this code yet (written on the fly) but it should hopefully get you started, if you have any issues let me know.

**Comment by kanertuncel** [comment]
Thanks, it worked. But I dont want it to loose it focused element. How should I do it? Because I'm replacing focused image on new dialog.

**Comment by kanertuncel** [comment]
by the way, @anthonyjb I'm trying to insert some editable contents from a template selector. You may look at screenshot I attached for what I'm trying to do. When I insert a region programmatically it allows me to edit paragraph. But the image is "static" and can't make it focused element. 
![screenshot_1](https://cloud.githubusercontent.com/assets/1259616/19193137/927eccb0-8cb1-11e6-9022-b5f1f7d875f6.png)

The html code I insert to editor is:

`<div class="section" data-editable data-name="content-text-image"><div class="align-left ce-element ce-element--type-image" style="background-image:url(/assets/newsletters/sample-image-300-200.png);width:300px;height:200px;" width="300" height="200" data-ce-size="w 300 × h 200"></div><p class="ce-element ce-element--type-text">Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p></div>`

**Comment by anthonyjb** [comment]
@kanertuncel the HTML you've provider here is what the editor would generate then it converted standard content into editable content. ContentTools/Edit wont parse editable state HTML as editable it will simply treat it as if it's standard content that needs parsing and so you wont get the result you are expecting.

**Comment by kanertuncel** [comment]
Hi @anthonyjb,
Thank you very much! I changed my html code into this and it worked. Now content I added programatically is editable. 

`<div class=\"section\" data-editable data-name=\"content-text-image\"><img src='/assets/newsletters/sample-image-300-200.png' width='300' height='200' class='align-left' /><p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p></div>`
